src/utils/mtev_hash.c

```diff
@@ -189,7 +228,33 @@ struct ck_hs_map {
   void **entries;
 };
 
+static void 
+mtev_hash_set_lock_mode_funcs(mtev_hash_table *h, mtev_hash_lock_mode_t lock_mode) 
+{
+  switch (lock_mode) {
+  case MTEV_HASH_LOCK_MODE_NONE:
+    h->lock = &none_lock;
+    h->unlock = &none_unlock;
+    break;
+  case MTEV_HASH_LOCK_MODE_MUTEX:
+    pthread_mutex_init(&h->locks.hs_lock, NULL);
+    h->lock = &mutex_lock;
+    h->unlock = &mutex_unlock;
```

Comment:
do we need to worry about where a corresponding `pthread_mutex_destroy` would go?